src/main/java/com/aws/greengrass/componentmanager/builtins/S3Downloader.java

```diff
@@ -111,15 +127,29 @@ public Long getDownloadSize() throws InterruptedException, PackageDownloadExcept
         // Parse artifact path
         String key = s3ObjectPath.key;
         String bucket = s3ObjectPath.bucket;
+        S3Client regionClient = getRegionClientForBucket(bucket);
```

Comment:
this can fail, can't it? Why is this removed from the `try`?

Reply:
If it fails it will throw either an InterruptedException or a proper PackageDownloadException. Moving out of try block so that we don't wrap it with another `PackageDownloadException(getErrorString("Failed to head artifact object from S3"), e)`